crates/cxx-qt-build/src/lib.rs

```diff
@@ -806,22 +865,102 @@ impl CxxQtBuilder {
+    ) {
+        if let Some(interface) = &self.public_interface {
+            // We automatically reexport all qt_modules and initializers from downstream dependencies
+            // as they will always need to be enabled in the final binary.
+            // However, we only reexport the headers and compile-time definitions of libraries that
+            // are marked as re-export.
+            let dependencies = dependencies::reexported_dependencies(interface, dependencies);
+
+            let initializers = interface
+                .initializers
+                .iter()
+                .cloned()
+                .chain(dependency_initializers);
+            let exported_include_prefixes =
+                dependencies::all_include_prefixes(interface, &dependencies);
+            let defines = dependencies::all_compile_definitions(Some(interface), &dependencies);
+
+            let manifest = Manifest {
+                name: crate_name(),
+                link_name: link_name()
+                    .expect("The links key must be set when creating a library with CXX-Qt-build!"),
+                defines,
+                initializers: initializers.collect(),
+                qt_modules: qt_modules.into_iter().collect(),
+                exported_include_prefixes,
+            };
+
+            let manifest_path = dir::crate_target().join("manifest.json");
+            let manifest_json = serde_json::to_string_pretty(&manifest)
+                .expect("Failed to convert Manifest to JSON!");
+            std::fs::write(&manifest_path, manifest_json).expect("Failed to write manifest.json!");
+            println!(
+                "cargo::metadata=CXX_QT_MANIFEST_PATH={}",
+                manifest_path.to_string_lossy()
+            );
+        }
+    }
+
+    fn qt_modules(&self, dependencies: &[Dependency]) -> HashSet<String> {
+        let mut qt_modules = self.qt_modules.clone();
+        for dependency in dependencies {
+            qt_modules.extend(dependency.manifest.qt_modules.iter().cloned());
+        }
+        qt_modules
+    }
+
+    fn write_interface_include_dirs(&self) {
+        if let Some(interface) = &self.public_interface {
+            for (header_dir, symlink) in &interface.exported_include_directories {
+                Self::symlink_directory(header_dir, dir::header_root().join(symlink))
+                    .unwrap_or_else(|_| {
+                        panic!(
+                            "Failed to create symlink `{}` for export_include_directory: {}",
+                            symlink,
+                            header_dir.to_string_lossy()
+                        )
+                    });
+            }
+        }
+    }
+
     /// Generate and compile cxx-qt C++ code, as well as compile any additional files from
     /// [CxxQtBuilder::qobject_header] and [CxxQtBuilder::cc_builder].
     pub fn build(mut self) {
+        // TODO: Clean the directory before we start building
```

Comment:
the code below runs clean, is this referring to something else ? it's not quite clear if this has been fixed or if there is a further clean ?

Reply:
Ah, right, I had to disable it for a while, it works now, so we can remove the comment :)